### PR TITLE
1.18 Update

### PR DESCRIPTION
**Pull request by Futuremappermydud**
Changes needed for 1.18, Some lights may have strange brightness values but I could not find the cause for this.

### CODE REVIEW

**Comment by Futuremappermydud** [comment]
Ignore build error as core mods arent on qpm lol

File: src/lighting/ChromaGradientController.cpp [hunk 1]
```diff
@@ -85,18 +85,24 @@ Sombrero::FastColor ChromaGradientController::AddGradient(ChromaEventData::Gradi
 }
 
 void Chroma::ChromaGradientController::Update() {
+    getLogger().debug("A");
+    return;
```
Comment (Fernthedev) [inline src/lighting/ChromaGradientController.cpp:89]: Remove debug logs and `return`

File: src/lighting/environment_enhancements/GameObjectTrackController.cpp [hunk 2]
```diff
@@ -139,7 +139,7 @@ void Chroma::GameObjectTrackController::Update() {
 
     auto posCalc = position ? position.value() * _noteLinesDistance : 0;
 
-    if (position && Sombrero::FastVector3(transform->get_position()) != (posCalc))
+    if (position && Sombrero::FastVector3(transform->get_position()) != ((UnityEngine::Vector3)posCalc))
```
Comment (Fernthedev) [inline src/lighting/environment_enhancements/GameObjectTrackController.cpp:142]: Change this to `Sombrero::FastVector3`

File: src/colorizer/LightColorizer.cpp [hunk 3]
```diff
@@ -261,9 +262,110 @@ void LightColorizer::SetSOs(std::vector<Sombrero::FastColor> const& colors) {
 void LightColorizer::Refresh() {
     auto const& colors = getColor();
     SetSOs(colors);
+
```
Comment (Fernthedev) [inline src/colorizer/LightColorizer.cpp:265]: Change `UnityEngine::Color` to `Sombrero::FastColor` and use `Alpha()` instead of `ColorExtensions

**Review by Fernthedev — CHANGES_REQUESTED** [review]

File: src/lighting/ChromaGradientController.cpp (hunk 1 above)
Comment (Futuremappermydud) [reply src/lighting/ChromaGradientController.cpp:89]: I forgot I added this to temporarily fix a crash 🤦

**Pull request closed without merging**